docs/layer2_account.md

```diff
@@ -0,0 +1,41 @@
+# Layer2 account
+
+Layer2 account can represent as: `(ID, L1 script)`
+
+* ID is an increased number.
+* L1 script, a CKB core data structure, unique for each layer2 account.
+
+The `Script` is defined as follows which is also a CKB core data structure:
+
+```
+/* https://github.com/nervosnetwork/ckb/blob/develop/util/types/schemas/blockchain.mol */
+table Script {
+    code_hash:      Byte32,
+    hash_type:      byte,
+    args:           Bytes,
+}
+```
+
+CKB uses this structure to indicate which script to load when verifying an L1 transaction. On Layer2, we reuse the structure:
+
+* code_hash - code_hash pointing to an executable binary that verifies a layer 2 transaction. For example: if a layer2 account represents an EVM contract, its `script.code_hash` is pointing to the binary that can verify the EVM transaction.
+* hash_type - hash_type affects how CKB loads the script binary, in layer2 account, it is fixed to `HashType::Type`.
+* args - args is used to set script initial args. We set the first 32 bytes to `rollup_script_hash` to distinguish accounts of different rollups.
+
+A layer2 account is whether an EOA or a contract is determined by its `code_hash`. If the script assumes the account is always a receiver of transactions then the account is a contract, otherwise, if the script assumes the account is always a sender of transactions then the account is an EOA. The script's code should perform some contract logic check or signature check based on the assumption.
```

Comment:
```suggestion
A layer2 account is whether an EOA or a contract determined by its `code_hash`. If the script assumes the account is always a receiver of transactions then the account is a contract. Otherwise, if the script assumes the account is always a sender of transactions then the account is an EOA. The script's code should perform some contract logic check or signature check based on the assumption.
```